src/pyFAI/gui/widgets/FitParamView.py

```diff
@@ -306,6 +306,9 @@ def __init__(self, parent, label, internalUnit, displayedUnit=None):
         if self._iconConstraintNoMax is None:
             self._iconConstraintNoMax = icons.getQIcon("pyfai:gui/icons/constraint-no-max")
 
+        displayedUnit.changed.connect(self.changeUnit)
+        self.changeUnit()  # enforcee the relabeling if needed
```

Comment:
Let's keep it private
```suggestion
        displayedUnit.changed.connect(self.__unitChanged)
        self.__unitChanged()  # enforce the relabeling if needed
```

Why passing label to the constructor since it is fully controlled by the displayed unit?